### ISSUE TITLE
Line number indicator uses wrong face for lines starting with TAB

### ISSUE THREAD

**Issue by theSuess** [meow-edit/meow#545]
When using `meow-line` in a file indented with `TAB` instead of `SPACE`, the face used to display line number indicators is not the same for indented lines. This messes with spacing and line height

![meow-indicator-wrong](https://github.com/meow-edit/meow/assets/1364576/b90aefc4-0ea5-4fbf-828b-e5f173ec3525)

**Comment by theSuess** [comment]
I'm using the following config:
```
    ;; Main typeface
    (set-face-attribute 'default nil :family "FantasqueSansM Nerd Font Mono" :height 140)
    ;; Set the fixed pitch face (monospace)
    (set-face-attribute 'fixed-pitch nil :family "FantasqueSansM Nerd Font Mono")
    ;; Set the variable pitch face
    (set-face-attribute 'variable-pitch nil :family "Atkinson Hyperlegible")
    ;; modeline
    (set-face-attribute 'mode-line nil :family "FantasqueSansM Nerd Font Mono" :height 140)
    (set-face-attribute 'mode-line-inactive nil :family "FantasqueSansM Nerd Font Mono" :height 140)
```
Let me know if you need any more information.

I'll try to reproduce this with a minimal configuration after work and share my findings here as well

**Comment by DogLooksGood** [comment]
I think you are missing a global font configuration, which should refers to "FantasqueSansM Nerd Font Mono" in your case.  
```
(add-to-list 'default-frame-alist '('font . "FantasqueSansM Nerd Font Mono-14"))
(set-frame-font "FantasqueSansM Nerd Font Mono-14")
```

**Comment by theSuess** [comment]
I don't think this has anything to do with the font. The minimal `init.el` reproduces this issue without any custom font configuration (emacs 29 from the fedora package repo).

I've tried the global font configuration as well, but it does not have any effect

**Comment by DogLooksGood** [comment]
![image](https://github.com/meow-edit/meow/assets/11796018/95d2c483-e763-4f53-a1e2-0f187ec8a74f)

`tab-width` doesn't affect anything in my case. What's the output of `M-x describe-face meow-position-highlight-number`?

It should be something like this
```
           Family: unspecified
          Foundry: unspecified
            Width: unspecified
           Height: unspecified
           Weight: unspecified
            Slant: unspecified
       Foreground: #202020
DistantForeground: #A0A0A0
       Background: unspecified
        Underline: unspecified
         Overline: unspecified
   Strike-through: unspecified
              Box: unspecified
          Inverse: unspecified
          Stipple: unspecified
             Font: unspecified
          Fontset: unspecified
           Extend: unspecified
          Inherit: default
 ```

**Comment by DogLooksGood** [comment]
Okay, I think I can reproduce it.

**Comment by DogLooksGood** [comment]
Fixed with 6a12dc4

**Comment by theSuess** [comment]
Thanks for the great package and such a quick fix!
